cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -116,12 +116,14 @@ func CreateStaticPodFiles(manifestDir string, cfg *kubeadmapi.ClusterConfigurati
 			return errors.Errorf("couldn't retrieve StaticPodSpec for %q", componentName)
 		}
 
-		// writes the StaticPodSpec to disk
-		if err := staticpodutil.WriteStaticPodToDisk(componentName, manifestDir, spec); err != nil {
-			return errors.Wrapf(err, "failed to create static pod manifest file for %q", componentName)
+		if !staticpodutil.CheckStaticPodExists(componentName, manifestDir) {
```

Comment:
We could create a check-or-create function which would help to refine the code, what do you think?

Reply:
+1 on me i think this PR implements check-or-create function. wdyt?

Reply:
It creates the manifests file only if it doesnot exist. if the manifests file exist it uses the same.